model/src/main/kotlin/SemanticVersionRange.kt

```diff
@@ -0,0 +1,182 @@
+/*
+ * Copyright (C) 2021 Bosch.IO GmbH
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+package org.ossreviewtoolkit.model
+
+import com.vdurmont.semver4j.Semver
+
+import java.time.LocalDate
+import java.time.Period
+
+/**
+ * A data class to represent a range of semantic versions.
+ *
+ * The bounds of the range are defined by [Semver] objects. For a version object or a string referencing a version it
+ * is possible to check whether it is contained in the range. Ranges can be created either by specifying the bounds
+ * or in relation to a given base version and a level or "granularity". In the latter case, the range contains the
+ * base version and all versions that differ from it only to a certain degree, e.g. in the patch level.
+ *
+ * This class also offers some support for [Calendar versioning][https://calver.org/] (CalVer), which is used by some
+ * tools ORT has to deal with. A CalVer is a semantic version, but the components of the version are derived from the
```

Comment:
> A CalVer is a semantic version

Are you sure? While I agree that CalVer *can* be used so it's also a SemVer, I believe it's usually not. That's because with CalVer you simply use the date of the release, independently of whether breaking changes have been introduced. Of course, a project using CalVer could decide to not do any releases with breaking changes within the same month. But then that's a project decision; CalVer by itself would allow that.

See also the example of [Teradata](https://calver.org/#teradata) where "the library maintainers have crafted a resourceful hybrid of semantic versioning and calendar versioning".

Reply:
Probably just a wording issue. What I meant is that a CalVer has the same components than a SemVer, and after a range of validity has been determined (e.g. based on a period), comparisons can be done in the same way to check whether a specific version is included.

Reply:
The `SemanticVersionRange` class has been dropped.